File: examples/01_basic_mapping.py
"""Example 1: Basic file-to-file mapping.

Maps a messy CRM export to a clean ERP customer schema.
This is the most common use case — you have data from one system
and need to figure out which columns correspond to another.
"""

import infermap

# Map CRM columns to ERP columns
result = infermap.map("data/crm_export.csv", "data/erp_customers.csv")

# Show what matched
print("=== Column Mappings ===\n")
for m in result.mappings:
    print(f"  {m.source:20s}  ->  {m.target:20s}  (confidence: {m.confidence:.3f})")
    print(f"    Reasoning: {m.reasoning}\n")

# Show what didn't match
if result.unmapped_source:
    print(f"Unmapped source columns: {result.unmapped_source}")
if result.unmapped_target:
    print(f"Unmapped target columns: {result.unmapped_target}")

# Show warnings
for w in result.warnings:
    print(f"WARNING: {w}")

print(f"\nCompleted in {result.metadata.get('elapsed_seconds', '?')}s")

File: examples/typescript/01_basic_mapping.ts
// 01 — Basic mapping: plain records → plain records.
//
// Run: npx tsx examples/typescript/01_basic_mapping.ts
//
// Shows the simplest path — pass two arrays of record objects to map()
// and let infermap infer columns, dtypes, and the optimal field pairing.

import { map, mapResultToJson } from "infermap";

const crmRecords = [
  { cust_id: "C001", fname: "John", lname: "Doe", email_addr: "john@acme.com", tel: "555-0100" },
  { cust_id: "C002", fname: "Jane", lname: "Smith", email_addr: "jane@globex.com", tel: "555-0200" },
  { cust_id: "C003", fname: "Bob", lname: "Johnson", email_addr: "bob@initech.com", tel: "555-0300" },
];

const erpRecords = [
  { customer_id: "", first_name: "", last_name: "", email: "", phone: "" },
];

const result = map(
  { records: crmRecords, sourceName: "crm" },
  { records: erpRecords, sourceName: "erp" }
);

console.log("== Mappings ==");
for (const m of result.mappings) {
  console.log(`  ${m.source.padEnd(12)} → ${m.target.padEnd(14)} (conf=${m.confidence.toFixed(3)})`);
}

if (result.unmappedSource.length > 0) {
  console.log(`Unmapped source: ${result.unmappedSource.join(", ")}`);
}
if (result.unmappedTarget.length > 0) {
  console.log(`Unmapped target: ${result.unmappedTarget.join(", ")}`);
}

// Full JSON output (report shape, rounded confidences):
// console.log(mapResultToJson(result));
void mapResultToJson;
